### PR TITLE
Build and Deploy a Container to GH Package Reg

### PR DESCRIPTION
**Pull request by kitplummer**
Adds a container build to the repo, deploying to the repo's package registry.

### CODE REVIEW

**Comment by kitplummer** [comment]
If passes, will about GHA to build containers only on `main`

**Comment by kitplummer** [comment]
NOTE: documentation for running container, will follow in subsequent PRs as we begin to test and evaluate the operation of the COD services from containers.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+# --more of a convenience when testing locally.
+COPY start_node.sh /app/
+COPY package*.json /app/
+COPY dist /app/dist
+# update each dependency in package.json to the latest version
+RUN npm install
+
+RUN chmod +x /app/start_node.sh
+ENTRYPOINT ["/app/start_node.sh"]
+CMD ["autov"]
```
Comment (ajfabbri) [inline Dockerfile:26]: Note: will use `config.json` by default but you can also pass the config filename as the first CLI arg if needed in the future.

**Review by ajfabbri — APPROVED** [review]
LGTM

File: Dockerfile (hunk 1 above)
Comment (kitplummer) [reply Dockerfile:26]: Yep, the container will allow you to specify any config file name, as long as it matches the one specified in the `-v` volume arg.  

For example (sorry for the long command):
```
docker run --network host -it --rm -p 5353:5353 -p 8082:8082 -v /var/run/dbus:/var/run/dbus -v /var/run/avahi-daemon/socket:/var/run/avahi-daemon/socket -v $PWD/autov-config.json:/app/config.json kit/m1-test node dist/autov-cod/index.js config.json
```

Comment (ajfabbri) [reply Dockerfile:26]: do you need a `--` after `index.js`, so the arg doesn't go to `node`?

**Pull request merged**